Configuracion de las herramientas.

MarinoFajardo opened the pull request:
En el fichero de configuración deno.js aparece al final una sección task, en esta sección se añaden las tareas que puede realizar el task runner. Hay que tener cuidado cuando incluimos una tarea, ya que se usa un subset de sh/bash para ejecutar las tareas.
Este subset en estos momentos no soporta los glob, aunque nos indican que lo incorporarán en el futuro.
En el fichero de configuración de Deno, se ha puesto la opción allowJs a false, por lo que los ficheros con extensión javascript no están permitidos. Las demás opciones se han dejado por defecto.

* Closes #16 porque incluye un gestor de tareas.

MarinoFajardo committed:
closes #21 con la configuración de Deno
Se ha creado el archivo de configuración de deno y se ha añadido la orden para comprobar sintaxis.

pedrooot reviewed (approved):
Todo bien 👍🏻. Ha sido útil revisar este PR, no sabía que deno se puede usar así!🥶🥳

JJ commented on deno.json:
```diff
@@ -0,0 +1,32 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
```
En principio no vamos a permitir JS, sólo TypeScript, ¿no? ¿O es por alguna otra razón?

JJ commented on deno.json:
```diff
@@ -0,0 +1,32 @@
+  "lint": {
+    "files": {
+      "include": ["src/"],
+      "exclude": ["src/tests/"]
```
¿Por qué? Habría que hacer linting en todo.

JJ commented on deno.json:
```diff
@@ -0,0 +1,32 @@
+    "rules": {
+      "tags": ["recommended"],
+      "include": ["ban-untagged-todo"],
+      "exclude": ["no-unused-vars"]
```
¿Por qué? Yo dejaría todas las que hay por defecto.

JJ commented on deno.json:
```diff
@@ -0,0 +1,32 @@
+      "exclude": ["no-unused-vars"]
+    }
+  },
+  "fmt": {
```
¿Esto qué hace?

JJ reviewed (changes requested):
Me parece bien en principio, pero 
1. habría que documentarlo en el README.md #29 
2. Justificar las diferentes elecciones, no usar simplemente el ejemplo
3. Usar todas las tareas que efectivamente se soliciten. `fmt` no estoy tan seguro. Si se quiere, debería crearse un issue que lo comente.

MarinoFajardo committed:
Avanza #29 documentando la elección de las herramientas
Se ha incluido el proceso de elección de las herramientas en el README

MarinoFajardo committed:
Actualización de la configuración de Deno
Se han dejado todas las opciones por defecto excluyendo el uso de archivos con extension javascript

MarinoFajardo committed:
Corregida herrata allowJS

JJ reviewed (approved):
🥇

JJ committed:
Merge branch 'main' into herramientas
